src/Orleans/Runtime/SiloStatus.cs

```diff
@@ -41,5 +41,13 @@ public static bool IsTerminating(this SiloStatus siloStatus)
         {
             return siloStatus == SiloStatus.ShuttingDown || siloStatus == SiloStatus.Stopping || siloStatus == SiloStatus.Dead;
         }
+
+        /// <summary>
+        /// Return true if silo is terminating or missing.
+        /// </summary>
+        public static bool IsUnavailable(this SiloStatus siloStatus)
+        {
+            return siloStatus.IsTerminating() || siloStatus == SiloStatus.None;
```

Comment:
So it's fine if this returns true for silos which we have not been informed about via membership yet?